src/components/WarningModal.tsx

```diff
@@ -0,0 +1,32 @@
+import React from "react";
+
+import { Modal, Button } from "react-bootstrap";
+
+const WarningModal: React.FC<{
+  title?: string;
+  message?: string;
+  show: boolean;
+  onDecline: () => void;
+  onConfirm: () => void;
+}> = (props) => {
```

Comment:
destructure props here and give initial values to title and message when they aren't mandatory.